cmd/ontap-cvo.go

```diff
@@ -232,10 +301,36 @@ func checkIfS3UserExists(mgmInfo managementInfo, svmUuid string, onPremName stri
 	// Build the request
 	urlString := "https://" + mgmInfo.managementIP + "/api/protocols/s3/services/" + svmUuid + "/users/" + onPremName
 	statusCode, _ := performHttpGet(mgmInfo.username, mgmInfo.password, urlString)
-	if statusCode == 200 {
+	if statusCode != 200 {
+		klog.Errorf("Error when checking if user exists:") // TODO add error message
+		return false
+	}
+	return true
+}
+
+/*
+This will check for the existence of an S3 bucket
+https://docs.netapp.com/us-en/ontap-restapi/ontap/get-protocols-s3-services-buckets-.html
+Requires: managementInfo, svm.uuid and the bucketName
+https://docs.netapp.com/us-en/ontap-restapi/ontap/protocols_s3_services_svm.uuid_buckets_endpoint_overview.html#retrieving-all-fields-for-all-s3-buckets-of-an-svm
+^ is the best we can do, given that we cannot search a bucket by its name (can search by UUID though)
+We'd need to re-use that hash function here when looking too.
+Returns true if it does exist
+*/
+func checkIfS3BucketExists(mgmInfo managementInfo, svmUuid string, requestedBucket string) bool {
+	// Build the request
+	urlString := "https://" + mgmInfo.managementIP + "/api/protocols/s3/services/" + svmUuid + "/buckets"
+	statusCode, _ := performHttpGet(mgmInfo.username, mgmInfo.password, urlString)
+	// statusCode, response := performHttpGet(mgmInfo.username, mgmInfo.password, urlString)
```

Comment:
Is this commented line necessary? Seems almost identical to the above one

Reply:
nope its not, was a relic